firstpartydata/first_party_data_test.go

```diff
@@ -0,0 +1,836 @@
+					t.Errorf("Unable to unmarshal site.content.data: %s", fileName)
+				}
+				reqFPD[siteContentDataKey] = siteConData
+			}
+
+			reqFPDAppContentData := fpdFile.FirstPartyData[appContentDataKey]
+			if len(reqFPDAppContentData) > 0 {
+				var appConData []openrtb2.Data
+				err = json.Unmarshal(reqFPDAppContentData, &appConData)
+				if err != nil {
+					t.Errorf("Unable to unmarshal app.content.data: %s", fileName)
+				}
+				reqFPD[appContentDataKey] = appConData
+			}
+
+			reqFPDUserData := fpdFile.FirstPartyData[userDataKey]
+			if len(reqFPDUserData) > 0 {
+				var userData []openrtb2.Data
+				err = json.Unmarshal(reqFPDUserData, &userData)
+				if err != nil {
+					t.Errorf("Unable to unmarshal app.content.data: %s", fileName)
+				}
+				reqFPD[userDataKey] = userData
+			}
+			if fpdFile.BiddersFPD == nil {
+				fpdFile.BiddersFPD = make(map[openrtb_ext.BidderName]*openrtb_ext.ORTB2)
+				fpdFile.BiddersFPD["appnexus"] = &openrtb_ext.ORTB2{}
+			}
+
+			resultFPD, errL := ResolveFPD(&inputReq, fpdFile.BiddersFPD, reqExtFPD, reqFPD, []string{"appnexus"})
+
+			if len(errL) == 0 {
+				assert.Equal(t, inputReq, inputReqCopy, "Original request should not be modified")
+
+				bidderFPD := resultFPD["appnexus"]
+
+				if bidderFPD.Site != nil && len(bidderFPD.Site.Ext) > 0 {
+					resSiteExt := bidderFPD.Site.Ext
+					expectedSiteExt := outputReq.Site.Ext
+					bidderFPD.Site.Ext = nil
+					outputReq.Site.Ext = nil
+					assert.JSONEq(t, string(expectedSiteExt), string(resSiteExt), "site.ext is incorrect")
+
+					assert.Equal(t, outputReq.Site, bidderFPD.Site, "Site is incorrect")
+				}
+				if bidderFPD.App != nil && len(bidderFPD.App.Ext) > 0 {
+					resAppExt := bidderFPD.App.Ext
+					expectedAppExt := outputReq.App.Ext
+					bidderFPD.App.Ext = nil
+					outputReq.App.Ext = nil
+					assert.JSONEq(t, string(expectedAppExt), string(resAppExt), "app.ext is incorrect")
+
+					assert.Equal(t, outputReq.App, bidderFPD.App, "App is incorrect")
+				}
+				if bidderFPD.User != nil && len(bidderFPD.User.Ext) > 0 {
+					resUserExt := bidderFPD.User.Ext
+					expectedUserExt := outputReq.User.Ext
+					bidderFPD.User.Ext = nil
+					outputReq.User.Ext = nil
+					assert.JSONEq(t, string(expectedUserExt), string(resUserExt), "user.ext is incorrect")
+
+					assert.Equal(t, outputReq.User, bidderFPD.User, "User is incorrect")
+				}
+			} else {
+				for i := range fpdFile.ValidationErrors {
+					assert.Contains(t, errL[i].Error(), fpdFile.ValidationErrors[i], "Incorrect first party data warning message")
+				}
+			}
+
+		}
+	}
+}
+
+func TestExtractFPDForBidders(t *testing.T) {
+
+	if specFiles, err := ioutil.ReadDir("./tests/extractfpdforbidders"); err == nil {
+		for _, specFile := range specFiles {
+			fileName := "./tests/extractfpdforbidders/" + specFile.Name()
+			fpdFile, err := loadFpdFile(fileName)
+
+			if err != nil {
+				t.Errorf("Unable to load file: %s", fileName)
+			}
+
+			var resRequest openrtb2.BidRequest
```

Comment:
Please call this `expectedRequest` or something like that as `resRequest` sounds more like this is the request that you get after you have processed it which this isn't. This is the expected output request specified in the JSON file